core/src/main/java/fr/sncf/osrd/conflicts/IncrementalPath.kt

```diff
@@ -36,25 +35,13 @@ class PathFragment(
         }
 }
 
-/** A marker type for Offset<Path> */
-sealed interface Path
-
 /** A marker type for Offset<TravelledPath> */
 sealed interface TravelledPath
 
 interface IncrementalPath {
     fun extend(fragment: PathFragment)
 
-    // TODO: implement trimming
-    fun setLastRequiredBlock(blockIndex: Int)
-
-    fun setLastRequiredRoute(routeIndex: Int)
-
-    fun trim()
-
-    val beginZonePathIndex: Int
-    val beginBlockIndex: Int
-    val beginRouteIndex: Int
+    fun clone(): IncrementalPath
```

Comment:
I think the current implementation is _alright_, but I can also see a cheaper structural sharing opportunity: instead of having separate linked lists for each field, IncrementalPath itself can be a linked list, where property accesses to fields before the current node's range are delegated to the next node